### ISSUE TITLE
[BUDI-7236] Navigation Link Roles not working as expected

### ISSUE THREAD

**Issue by tglynx** [Budibase/budibase#11038]
**Checklist**
- [x] I have searched budibase discussions and github issues to check if my issue already exists
This issue may be related to #7917, however, in that case a navigation link is not visible when assigned to a custom role directly. 
Here a navigation link is visible to all members of the Basic role when the link is assigned to a role that inherits from 'Basic' (as an example). This may also be a security issue.

**Hosting**
- Self
  - Method: docker compose
  - Budibase Version: 2.6.20
  - App Version: 2.6.20

**Describe the bug**
I have a custom role with the following setup: 
Name: Member
Inherits Role: Basic
Base Permission: Read / Write

As per my understanding the 'Member' role inherits all the permissions of the 'Basic' role which is fine. 
However, if i assign the 'Member' Role to a 'Navigation Link' this link can also be seen by members of the 'Basic' role.
I would assume that just members of the 'Member' role should be able to see the link.

**To Reproduce**
1. Create a custom role based on 'Basic' with Read/write permissions
2. Assign the new role to a navigation link
3. Log in with a member of a lower role (Basic) 
4. See error (link assigned to higher role is visible)

**Expected behavior**
I would expect that a Basic User can only see navigtion links that have been assigned to Basic or Public roles.
I would expect that a navigation link that has a custom role assigned is only visible to members of this custom role or roles that inherit from this custom role.

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser Chrome
 - Version 114.0.5735.133

<sub>[BUDI-7236](https://linear.app/budibase/issue/BUDI-7236/navigation-link-roles-not-working-as-expected)</sub>

**Comment by tglynx** [comment]
I add some additional information as i really think this should be fixed and to try to display the problem as easy as possible!

I have created a custom role called 'Member' that inherits from 'Power' (same behaviour observed when using Basic)
<img width="346" alt="Bildschirmfoto 2023-08-19 um 11 55 13" src="https://github.com/Budibase/budibase/assets/30650435/0b4fe4fe-44fb-49d0-8cbf-432f777de1ed">

In this example we have look at two navigational links (/profile and /debug): 
<img width="827" alt="Bildschirmfoto 2023-08-19 um 11 54 11" src="https://github.com/Budibase/budibase/assets/30650435/8a6a6e00-93e8-4e8d-8246-25d0e34efa49">
<img width="839" alt="Bildschirmfoto 2023-08-19 um 11 53 48" src="https://github.com/Budibase/budibase/assets/30650435/60d667ce-f9d6-4718-b3a2-fb75ceec4625">

When a user with the Basic role assigned is logged in we would expect the following: 
- /profile -> visible (basic role assigned)
- /debug -> not visible (member role assigned)

In fact we observe the following:
- /profile -> visible (OK)
- /debug -> visible (why? - should only be visible for members of the 'Member' role! Even 'Power' Role members should not be able to see this!)

When a user with the Member role assigned is logged in we would expect the following;
- /profile -> visible (Member role assigned that is inheriting Power which is higher than Basic)
- /debug -> visible (Member role assigned)

In fact we observe the following: 
- /profile -> not visible (why?)
- /debug -> visible (OK)

This is really broken for custom roles :) And we need custom roles!

In short: 
The public role (everyone) can see all menu items that have been assigned to a custom role!
The custom role only sees menu items that have been assigned to that role explicitly!

Hope this helps to understand the problem and to fix this. It currently prevents us to develop our solution further ...

**Comment by tglynx** [comment]
A lot has been done in rbac lately!
However, this bug is still present in version 2.10.12

**Comment by mike12345567** [comment]
Hi @tglynx - this should be fixed in Budibase 2.11.X.

**Comment by tglynx** [comment]
After quite some testing i can confirm it is working as expected now. Thanks!
While testing i found a new bug related to RBAC when changing a custom role. I will open a new issue for that... 
In short: When changing a custom role (for example the 'Inherits Role' or 'Base Permissions' properties all assignments to that role get set to Role 'NULL' (Links, Users, Screens).
